### PR TITLE
rpc: support NotaryRequest notifications for Notifications subsystem

### LINKED ISSUE #1960: Add subscription to notary payload pool events
If someone wants to react on notary payload (like analyze and complete it) he should be able to see this payload in the first place and at the moment there is no way to do that. We can add yet another subscription channel to our notification system and allow to receive notary payloads (filtered by signer).

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #1960.

### CODE REVIEW

File: pkg/rpc/response/events.go [hunk 1]
```diff
@@ -3,11 +3,20 @@ package response
 import (
 	"encoding/json"
 	"errors"
+
+	"github.com/nspcc-dev/neo-go/pkg/core/mempool"
```
Comment (roman-khimov) [inline pkg/rpc/response/events.go:7]: It's so weird importing it here, maybe we can move event definitions out?

File: pkg/rpc/response/events.go [hunk 2]
```diff
@@ -3,11 +3,20 @@ package response
...
 )
 
 type (
 	// EventID represents an event type happening on the chain.
 	EventID byte
+	// NotaryRequestEvent represents P2PNotaryRequest event either added or removed
+	// from notary payload pool.
+	NotaryRequestEvent struct {
+		Type          mempool.EventType         `json:"type"`
+		NotaryRequest *payload.P2PNotaryRequest `json:"notary_request"`
```
Comment (roman-khimov) [inline pkg/rpc/response/events.go:18]: Snaky.

File: pkg/core/mempool/event.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
+package mempool
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply pkg/core/mempool/event.go]: It was a separate package firstly, but it didn't look good, so I moved it to a separate file. Now it's a separate package again, but I'm not sure where to put it.

File: pkg/rpc/response/events.go (hunk 1 above)
Comment (AnnaShaleva) [reply pkg/rpc/response/events.go]: Moved to a `mempoolevent` package, although it's not really needed to create a separate package for it (there's no dependency cycle problem).

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**